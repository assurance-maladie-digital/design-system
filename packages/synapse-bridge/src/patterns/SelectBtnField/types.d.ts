export type SelectBtnValue = string | number | Array<string | number> | null;

export interface SelectBtnItem {
	text: string;
	value: string | number;
	unique?: boolean;
}
