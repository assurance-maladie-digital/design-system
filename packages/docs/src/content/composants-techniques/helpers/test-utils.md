---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
