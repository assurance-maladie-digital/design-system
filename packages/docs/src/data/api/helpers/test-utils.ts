import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
