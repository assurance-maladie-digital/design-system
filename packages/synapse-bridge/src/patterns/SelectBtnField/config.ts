export const config = {
	btn: {
		height: "auto",
		minHeight: "40px",
		activeClass: "text-white",
		class: "py-2 d-flex align-items-center",
	},
	icon: {
		class: "text-white flex-shrink-0 ml-1",
	},
};
