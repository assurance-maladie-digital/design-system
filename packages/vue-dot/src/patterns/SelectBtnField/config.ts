export const config = {
	btn: {
		height: 'auto',
		minHeight: '40px',
		activeClass: 'white--text',
		class: 'text-wrap text-left py-2'
	},
	icon: {
		class: 'white--text flex-shrink-0 ml-1'
	}
};
