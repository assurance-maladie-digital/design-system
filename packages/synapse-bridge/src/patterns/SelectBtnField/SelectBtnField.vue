<script lang="ts">
import { defineComponent } from 'vue';
import type { PropType } from 'vue';
import { customizable } from '@/mixins/customizable';
import { config } from './config';
import type { SelectBtnValue, SelectBtnItem } from './types';
import { mdiCheck } from '@mdi/js';
import { useTheme } from 'vuetify';

export default defineComponent({
	mixins: [customizable(config)],
	props: {
		modelValue: {
			type: [Number, String, Array] as PropType<SelectBtnValue>,
			default: null,
		},
		items: {
			type: Array as PropType<SelectBtnItem[]>,
			default: () => [],
		},
		label: {
			type: String,
			default: undefined,
		},
		multiple: {
			type: Boolean,
			default: false,
		},
		inline: {
			type: Boolean,
			default: false,
		},
		hint: {
			type: String,
			default: undefined,
		},
		error: {
			type: Boolean,
			default: false,
		},
		errorMessages: {
			type: Array as PropType<string[]>,
			default: undefined,
		},
		readonly: {
			type: Boolean,
			default: false
		}
	},
	watch: {
		modelValue: {
			handler(value: SelectBtnValue): void {
				if (value === null && this.multiple) {
					this.internalValue = [];

					return;
				}

				this.internalValue = value;
			},
			immediate: true,
			deep: true,
		},
	},
	emits: ['update:modelValue', 'update:error', 'update:error-messages'],
	data() {
		return {
			checkIcon: mdiCheck,
			internalValue: null as SelectBtnValue,
			darktheme: useTheme().current.value.dark as boolean,
		};
	},
	computed: {
		filteredItems(): SelectBtnItem[] {
			return this.items.filter((item) => {
				return item.value !== null && item.value !== undefined;
			});
		},
	},
	methods: {
		isSelected(value: number | string): boolean {
			if (this.multiple) {
				return Array.isArray(this.internalValue) && this.internalValue.includes(value);
			}
			return this.internalValue === value;
		},

		getIconStyles(item: SelectBtnItem): Record<string, string> {
			return {
				visibility: this.isSelected(item.value) ? 'visible' : 'hidden',
			};
		},

		getNewValue(item: SelectBtnItem): SelectBtnValue {
			if (this.multiple) {
				const typedValue = this.internalValue as Array<string | number>;

				// If the item is unique, remove all other items from the array
				const hasUniqueItemSelected = this.filteredItems.some(
					(filteredItem) => filteredItem.unique && typedValue.includes(filteredItem.value)
				);

				if (item.unique || hasUniqueItemSelected) {
					return [item.value];
				}

				// If the item is not already selected, add it to the array
				if (!typedValue.includes(item.value)) {
					return [...typedValue, item.value];
				}

				// If the item is already selected, remove it from the array
				return typedValue.filter((value) => value !== item.value);
			}

			// If the item is already selected, deselect it
			if (this.internalValue === item.value) {
				return null;
			}

			// Select the item
			return item.value;
	},

		toggleItem(item: SelectBtnItem): void {
			this.internalValue = this.getNewValue(item);

			this.$emit('update:error', false);
			this.$emit('update:error-messages', undefined);
			this.$emit('update:modelValue', this.internalValue);
		},
	},
});
</script>

<template>
	<div :class="{ 'vd-form-input': !inline }" class="vd-select-btn-field">
		<VBtnToggle
			v-bind="options.btnToggle"
			:model-value="internalValue"
			:multiple="multiple"
			:class="{ 'flex-column': !inline }"
			:aria-label="label"
			role="listbox"
			class="vd-select-btn-field-toggle d-flex flex-wrap text-primary"
		>
			<VBtn
				v-for="(item, index) in filteredItems"
				v-bind="options.btn"
				:key="index"
				:value="item.value"
				:disabled="readonly"
				:variant="isSelected(item.value) ? 'flat' : 'outlined'"
				:elevation="0"
				:color="error ? 'error' : 'primary'"
				:class="{
					'text-error': error && !isSelected(item.value),
					'justify-start': !isSelected(item.value),
					'justify-space-between': isSelected(item.value),
				}"
				:aria-selected="isSelected(item.value)"
				role="option"
				@click="toggleItem(item)"
			>
				<span class="text-body-1">
					{{ item.text }}
				</span>

				<VIcon
					v-bind="options.icon"
					v-if="isSelected(item.value) || inline"
					:style="getIconStyles(item)"
				>
					{{ checkIcon }}
				</VIcon>
			</VBtn>
		</VBtnToggle>

		<template v-if="errorMessages">
			<p
				v-for="(errorMessage, index) in errorMessages"
				:key="index"
				:class="darktheme ? 'theme--dark' : 'theme--light'"
				class="v-messages text-error px-3 mt-2 mb-0"
			>
				{{ errorMessage }}
			</p>
		</template>

		<p
			v-else-if="hint"
			:class="darktheme ? 'theme--dark' : 'theme--light'"
			class="v-messages px-3 mt-2 mb-0"
		>
			{{ hint }}
		</p>
	</div>
</template>

<style lang="scss" scoped>
@import "@cnamts/design-tokens/dist/tokens";

.vd-select-btn-field-toggle {
	background: none !important;

	&.flex-column .v-btn {
		border-radius: 4px !important;

		+ .v-btn {
			margin-top: 8px;
		}
	}

	.v-btn {
		background: #fff;
		border-width: 1px !important;

		&.v-btn--active::before {
			opacity: 0 !important;
		}

		:deep(.v-btn__content) {
			flex-shrink: 1 !important;
		}
	}

	&.theme--dark {
		.v-btn {
			background: #1e1e1e;
		}
	}

	&:not(.v-btn-toggle--group) .v-btn.v-btn {
		border-color: $vd-primary !important;

		&.text-error {
			border-color: $vd-error !important;
		}
	}

	// Disable the fade effect when the button is disabled
	:deep(.v-btn--disabled) {
		opacity: 1;
	}

	:deep(.v-btn--disabled.v-btn--variant-flat .v-btn__overlay) {
		opacity: 0;
	}
}
</style>
