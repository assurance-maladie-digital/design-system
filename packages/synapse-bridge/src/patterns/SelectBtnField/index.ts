import SelectBtnField from './SelectBtnField.vue';

export default SelectBtnField;
