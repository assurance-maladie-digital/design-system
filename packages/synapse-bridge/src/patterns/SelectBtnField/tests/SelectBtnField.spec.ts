import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import { vuetify } from "@tests/unit/setup";

import SelectBtnField from "../SelectBtnField.vue";

describe("SelectBtnField", () => {
	it("renders correctly", () => {
		const wrapper = mount(SelectBtnField, {
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders correctly with props", () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				hint: "Test",
				items: [
					{
						text: "Test",
						value: "test",
					},
					{
						text: "Test 2",
						value: "",
					},
					{
						text: "Test 3",
						value: "test3",
					}
				],
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('render correctly in multiple mode', () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				hint: "Test",
				items: [
					{
						text: "Test",
						value: "test",
					},
					{
						text: "Test 2",
						value: "",
					},
					{
						text: "Test 3",
						value: "test3",
					}
				],
				multiple: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('emits an update event when the value change in single mode', async () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				hint: "Test",
				items: [
					{
						text: "Test",
						value: "test",
					},
				],
			},
			global: {
				plugins: [vuetify],
			},
		});

		await wrapper.find('.v-btn').trigger('click');

		expect(wrapper.emitted()).toHaveProperty('update:modelValue');

		await wrapper.find('.v-btn').trigger('click');

		expect(wrapper.emitted('update:modelValue')).toEqual([['test'], [null]]);
	});

	it(`emits an array of values when the value changes in multiple mode`, async () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				hint: "Test",
				items: [
					{
						text: "Test",
						value: "test",
					},
					{
						text: "Test 2",
						value: "test2",
					},
					{
						text: "Test 3",
						value: "test3",
					},
				],
				multiple: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		await wrapper.find('.v-btn:nth-child(2)').trigger('click');
		await wrapper.find('.v-btn:nth-child(3)').trigger('click');
		await wrapper.find('.v-btn:nth-child(2)').trigger('click');

		expect(wrapper.emitted('update:modelValue')).toEqual([[['test2']], [['test2', 'test3']], [['test3']]]);
	});

	it(`display correctly with an error`, () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				hint: "Test",
				items: [
					{
						text: "Test",
						value: "test",
					},
					{
						text: "Test 2",
						value: "test2",
					},
					{
						text: "Test 3",
						value: "test3",
					},
				],
				error: true,
				errorMessages: ['Test'],
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it(`clear the others values when defined to unique`, async () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				items: [
					{
						text: "Test 1",
						value: "test1",
					},
					{
						text: "Test 2",
						value: "test2",
					},
					{
						text: "Other",
						value: "other",
						unique: true,
					},
				],
				multiple: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		await wrapper.find('.v-btn:nth-child(1)').trigger('click');
		await wrapper.find('.v-btn:nth-child(2)').trigger('click');
		await wrapper.find('.v-btn:nth-child(3)').trigger('click');
		await wrapper.find('.v-btn:nth-child(2)').trigger('click');

		expect(wrapper.emitted('update:modelValue')).toEqual(
			[[['test1']],[['test1', 'test2']], [['other']], [['test2']]]
		);
	});

	it (`display correctly in dark mode with an error`, () => {
		const DarkMode = {
			template: `
				<v-app>
					<v-theme-provider theme="dark">
						<div>
							<SelectBtnField v-bind="props" />
						</div>
					</v-theme-provider>
				</v-app>
			`,
			components: {
				SelectBtnField,
			},
			data() {
				return {
					props: {
						label: "Test",
						hint: "Test",
						items: [
							{
								text: "Test 1",
								value: "test1",
							},
							{
								text: "Test 2",
								value: "test2",
							},
						],
						error: true,
						errorMessages: ['Test'],
						multiple: true,
						inline: true,
					},
				};
			},

		};

		const wrapper = mount(DarkMode, {
			global: {
				plugins: [vuetify],
			},
		});

		wrapper.find('.v-btn:nth-child(1)').trigger('click');
		wrapper.find('.v-btn:nth-child(2)').trigger('click');

		expect(wrapper.html()).toMatchSnapshot();
	});

	it (`display correctly with in dark mode with an hint`, () => {
		const DarkMode = {
			template: `
				<v-app>
					<v-theme-provider theme="dark">
						<div>
							<SelectBtnField v-bind="props" />
						</div>
					</v-theme-provider>
				</v-app>
			`,
			components: {
				SelectBtnField,
			},
			data() {
				return {
					props: {
						label: "Test",
						hint: "Test",
						items: [
							{
								text: "Test 1",
								value: "test1",
							},
							{
								text: "Test 2",
								value: "test2",
							},
						],
						multiple: true,
						inline: true,
					},
				};
			},

		};

		const wrapper = mount(DarkMode, {
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('do not allow to select an item when the readonly prop is defined', async () => {
		const wrapper = mount(SelectBtnField, {
			props: {
				label: "Test",
				hint: "Test",
				items: [
					{
						text: "Test 1",
						value: "test1",
					},
					{
						text: "Test 2",
						value: "test2",
					},
				],
				readonly: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		await wrapper.find('.v-btn').trigger('click');
		expect(wrapper.emitted()).not.toHaveProperty('update:modelValue');
	});
});
