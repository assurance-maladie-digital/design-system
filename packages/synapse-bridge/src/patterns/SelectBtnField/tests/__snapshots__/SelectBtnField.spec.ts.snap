// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectBtnField > display correctly in dark mode with an error 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr">
  <div class="v-application__wrap">
    <div>
      <div data-v-044548c4="" class="vd-select-btn-field">
        <div data-v-044548c4="" class="v-btn-group v-theme--dark v-btn-group--density-default v-btn-toggle vd-select-btn-field-toggle d-flex flex-wrap text-primary" aria-label="Test" role="listbox"><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--dark v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center text-error justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test1"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 1</span><i data-v-044548c4="" class="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z mdi v-icon notranslate v-theme--dark v-icon--size-default text-white flex-shrink-0 ml-1" style="visibility: hidden;" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--dark v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center text-error justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test2"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 2</span><i data-v-044548c4="" class="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z mdi v-icon notranslate v-theme--dark v-icon--size-default text-white flex-shrink-0 ml-1" style="visibility: hidden;" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button></div>
        <p data-v-044548c4="" class="theme--dark v-messages text-error px-3 mt-2 mb-0">Test</p>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectBtnField > display correctly with an error 1`] = `
<div data-v-044548c4="" class="vd-form-input vd-select-btn-field">
  <div data-v-044548c4="" class="v-btn-group v-theme--light v-btn-group--density-default v-btn-toggle flex-column vd-select-btn-field-toggle d-flex flex-wrap text-primary" aria-label="Test" role="listbox"><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center text-error justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center text-error justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test2"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 2</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center text-error justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test3"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 3</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button></div>
  <p data-v-044548c4="" class="theme--light v-messages text-error px-3 mt-2 mb-0">Test</p>
</div>
`;

exports[`SelectBtnField > display correctly with in dark mode with an hint 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr">
  <div class="v-application__wrap">
    <div>
      <div data-v-044548c4="" class="vd-select-btn-field">
        <div data-v-044548c4="" class="v-btn-group v-theme--dark v-btn-group--density-default v-btn-toggle vd-select-btn-field-toggle d-flex flex-wrap text-primary" aria-label="Test" role="listbox"><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--dark v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test1"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 1</span><i data-v-044548c4="" class="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z mdi v-icon notranslate v-theme--dark v-icon--size-default text-white flex-shrink-0 ml-1" style="visibility: hidden;" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--dark v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test2"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 2</span><i data-v-044548c4="" class="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z mdi v-icon notranslate v-theme--dark v-icon--size-default text-white flex-shrink-0 ml-1" style="visibility: hidden;" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button></div>
        <p data-v-044548c4="" class="theme--dark v-messages px-3 mt-2 mb-0">Test</p>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectBtnField > render correctly in multiple mode 1`] = `
<div data-v-044548c4="" class="vd-form-input vd-select-btn-field">
  <div data-v-044548c4="" class="v-btn-group v-theme--light v-btn-group--density-default v-btn-toggle flex-column vd-select-btn-field-toggle d-flex flex-wrap text-primary" aria-label="Test" role="listbox"><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 2</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test3"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 3</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button></div>
  <p data-v-044548c4="" class="theme--light v-messages px-3 mt-2 mb-0">Test</p>
</div>
`;

exports[`SelectBtnField > renders correctly 1`] = `
<div data-v-044548c4="" class="vd-form-input vd-select-btn-field">
  <div data-v-044548c4="" class="v-btn-group v-theme--light v-btn-group--density-default v-btn-toggle flex-column vd-select-btn-field-toggle d-flex flex-wrap text-primary" role="listbox"></div>
  <!--v-if-->
</div>
`;

exports[`SelectBtnField > renders correctly with props 1`] = `
<div data-v-044548c4="" class="vd-form-input vd-select-btn-field">
  <div data-v-044548c4="" class="v-btn-group v-theme--light v-btn-group--density-default v-btn-toggle flex-column vd-select-btn-field-toggle d-flex flex-wrap text-primary" aria-label="Test" role="listbox"><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 2</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button><button data-v-044548c4="" type="button" class="v-btn v-btn--flat v-theme--light v-btn--density-default elevation-0 v-btn--size-default v-btn--variant-outlined py-2 d-flex align-items-center justify-start" style="height: auto; min-height: 40px;" activeclass="text-white" aria-selected="false" role="option" value="test3"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator=""><span data-v-044548c4="" class="text-body-1">Test 3</span>
      <!--v-if--></span>
      <!---->
      <!---->
    </button></div>
  <p data-v-044548c4="" class="theme--light v-messages px-3 mt-2 mb-0">Test</p>
</div>
`;
